src/main/java/de/javakaffee/kryoserializers/protobuf/ProtobufSerializer.java

```diff
@@ -0,0 +1,68 @@
+package de.javakaffee.kryoserializers.protobuf;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.google.protobuf.GeneratedMessage;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class ProtobufSerializer<T extends GeneratedMessage> extends Serializer<T> {
+
+    private static final int NULL = -1;
+
+    @Override
+    public void write(Kryo kryo, Output output, T protobufMessage) {
+        // If our protobuf is null
+        if (protobufMessage == null) {
+            // Write our special null value
+            output.writeByte(NULL);
+            output.flush();
+
+            // and we're done
+            return;
+        }
+
+        // Otherwise serialize protobuf to a byteArray
+        byte[] bytes = protobufMessage.toByteArray();
+
+        // Write the length of our byte array
+        output.writeInt(bytes.length + 1, true);
+
+        // Write the byte array out
+        output.writeBytes(bytes);
+        output.flush();
+    }
+
+    @Override
+    public T read(Kryo kryo, Input input, Class<T> type) {
+        // Read the length of our byte array
+        int length = input.readInt(true);
+
+        // If the length is equal to our special null value
+        if (length == NULL) {
+            // Just return null
+            return null;
+        }
+        // Otherwise read the byte array length
+        byte[] bytes = input.readBytes(length - 1);
+        try {
+            // Deserialize protobuf
+            Method method = type.getMethod("parseFrom", byte[].class);
```

Comment:
Updated to cache the reflection call, and expanded the tests a bit to verify it works as expected across multiple protobuf instances/types.